src/borg/archiver.py

```diff
@@ -327,15 +327,21 @@ def do_check(self, args, repository):
             # also, there is no max_duration support in the archives check code anyway.
             self.print_error("--repository-only is required for --max-duration support.")
             return EXIT_ERROR
+        checker = None
+        if args.prefix is not None:
+            args.glob_archives = args.prefix + '*'
+        if not args.repo_only:
+            try:
+                checker = ArchiveChecker(repository, repair=args.repair, archive=args.location.archive, first=args.first,
+                                         last=args.last, sort_by=args.sort_by or 'ts', glob=args.glob_archives,
+                                         verify_data=args.verify_data, save_space=args.save_space)
+            except Exception as exc:
+                self.print_error(str(exc))
+                return EXIT_WARNING
         if not args.archives_only:
             if not repository.check(repair=args.repair, save_space=args.save_space, max_duration=args.max_duration):
```

Comment:
the potential fundamental problem with first accessing the repo inside `ArchiveChecker.__init__` before doing `repository.check` is that we do not know whether we are working on valid data.

The repo check / repair makes sure that the fundamental data is ok (in memory and in repair mode also on disk).
The archives check / repair builds upon that (and assumes that lower level structures are ok).

So, guess there needs to be pretty much review before touching repo data before the repo has been checked.

Reply:
OK, and that's what I feared. Is there some way to verify only the chunks that contain the metadata needed to decrypt in `ArchiveChecker.__init__()`, so we know that the crypto & manifest & whatever other metadata is kosher, but don't spend 5 hours doing all of `repo/data/*`

Reply:
yes, a "miniature version" of the borg repo check, that only makes sure that 1 chunk is ok (which could be the manifest (id 0) or any other chunk). And then detect crypto based on that chunk.

that might work, needs checking...